### DISCUSSION TITLE
Ideas for getting race results as soon as telemetry is available?

### DISCUSSION THREAD

**Discussion by pesaventofilippo** [theOehrly/Fast-F1#425] [Q&A]
Telemetry from livetiming.formula1.com is usually available ~30 minutes after a race ends. The problem is not all information is given, for example the starting grid position for each driver is initially set to NaN and only updated some time after (1 or 2 hours maybe?)

Do you have any idea how we could get/calculate data like the starting grid position, or the end position (ClassifiedPosition) as soon as the other data is available?

For the ending position we could use the Position for every lap and pick the last one, but it only works for non-retired and non-lapped drivers

**Comment by theOehrly** [comment]
> Telemetry from livetiming.formula1.com is usually available ~30 minutes after a race ends. The problem is not all information is given, for example the starting grid position for each driver is initially set to NaN and only updated some time after (1 or 2 hours maybe?)

The source for the initially missing data is Ergast which takes longer to be updated.


> Do you have any idea how we could get/calculate data like the starting grid position, or the end position (ClassifiedPosition) as soon as the other data is available?
>
> For the ending position we could use the Position for every lap and pick the last one, but it only works for non-retired and non-lapped drivers

Calculating this information is possible in theory, at least partially. This is implemented for sprint qualifying, because Ergast provides not dara for that. But as you note, there are some things that we cannot account for then. 
The result on Ergast ist the official result that includes all penalties applied after the race as well. 
I don't really want to add the calculated result just to make a partially incorrect result available sooner. This would be very difficult to communicate to users and could lead to confusion.
